src/error.rs

```diff
@@ -25,119 +27,122 @@ impl actix_web::error::ResponseError for AybError {
 
 impl From<fernet::DecryptionError> for AybError {
     fn from(_cause: fernet::DecryptionError) -> Self {
-        AybError {
+        AybError::Other {
             message: "Invalid or expired token".to_owned(),
         }
     }
 }
 
 impl From<lettre::address::AddressError> for AybError {
     fn from(cause: lettre::address::AddressError) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("Invalid email address: {}", cause),
         }
     }
 }
 
 impl From<prefixed_api_key::BuilderError> for AybError {
     fn from(cause: prefixed_api_key::BuilderError) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("Error in prefixed API key builder: {}", cause),
         }
     }
 }
 
 impl From<prefixed_api_key::PrefixedApiKeyError> for AybError {
     fn from(cause: prefixed_api_key::PrefixedApiKeyError) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("Error parsing API token: {}", cause),
         }
     }
 }
 
 impl From<quoted_printable::QuotedPrintableError> for AybError {
     fn from(cause: quoted_printable::QuotedPrintableError) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("{:?}", cause),
         }
     }
 }
 
 impl From<rusqlite::Error> for AybError {
     fn from(cause: rusqlite::Error) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("{:?}", cause),
         }
     }
 }
 
 impl From<rusqlite::types::FromSqlError> for AybError {
     fn from(cause: rusqlite::types::FromSqlError) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("{:?}", cause),
         }
     }
 }
 
 impl From<string::FromUtf8Error> for AybError {
     fn from(cause: string::FromUtf8Error) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("{:?}", cause),
         }
     }
 }
 
 impl From<serde_json::Error> for AybError {
     fn from(cause: serde_json::Error) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("{:?}", cause),
         }
     }
 }
 
 impl From<std::str::Utf8Error> for AybError {
     fn from(cause: std::str::Utf8Error) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("{:?}", cause),
         }
     }
 }
 
 impl From<std::io::Error> for AybError {
     fn from(cause: std::io::Error) -> Self {
-        AybError {
+        AybError::Other {
             message: format!("IO error: {:?}", cause),
         }
     }
 }
 
 impl From<sqlx::Error> for AybError {
     fn from(cause: sqlx::Error) -> Self {
-        AybError {
-            message: format!("{:?}", cause),
+        match cause {
+            sqlx::Error::RowNotFound => Self::RecordNotFound,
```

Comment:
Could we catch it closer to the original location so we can include a message like "Entity not found" with an ID/slug?

Reply:
> Could we catch it closer to the original location so we can include a message like "Entity not found" with an ID/slug?

Yes, of course :)

> You could imagine including a type of record and string identifier to help with debugging :)

I'm not really sure I follow, something like this?
```rust
enum AybError {
    RecordNotFound { id: String, record_type: String },
    ...
}
```

Reply:
> I'm not really sure I follow, something like this?
```rust
enum AybError {
    RecordNotFound { id: String, record_type: String },
    ...
}

Exactly like that! :) I think it's OK that `id` is a `String` even if it's sometimes an int and sometimes a String, since it's just for helping figure out what lookup went wrong.